core/src/main/java/org/elasticsearch/index/query/GeoDistanceQueryParser.java

```diff
@@ -120,9 +120,6 @@ public GeoDistanceQueryBuilder fromXContent(QueryParseContext parseContext) thro
                 } else if (currentFieldName.endsWith(GeoPointFieldMapper.Names.LON_SUFFIX)) {
                     point.resetLon(parser.doubleValue());
                     fieldName = currentFieldName.substring(0, currentFieldName.length() - GeoPointFieldMapper.Names.LON_SUFFIX.length());
-                } else if (currentFieldName.endsWith(GeoPointFieldMapper.Names.GEOHASH_SUFFIX)) {
```

Comment:
Shouldn't we be able to remove `GeoPointFieldMapper.Names.GEOHASH_SUFFIX` entirely from the code base?